Sources/Workspace/Workspace.swift

```diff
@@ -1401,32 +1401,45 @@ extension Workspace {
 
     /// Removes the clone and checkout of the provided specifier.
     private func remove(specifier: RepositorySpecifier) throws {
-        guard var dependency = managedDependencies[specifier] else {
+        
+        guard let dependency = managedDependencies[specifier] else {
             fatalError("This should never happen, trying to remove \(specifier) which isn't in workspace")
         }
-
-        // If this dependency is based on a dependency, switch to that because
-        // we don't want to touch the editable checkout here.
-        //
-        // FIXME: This will remove also the data about the editable dependency
-        // and it will not be possible to "unedit" that dependency anymore.  To
-        // do that we need to persist the value of isInEditableState and also
-        // store the package names in managed dependencies, because it will not
-        // be possible to lookup these dependencies using their manifests as we
-        // won't have them anymore.
-        // https://bugs.swift.org/browse/SR-3689
-        if let basedOn = dependency.basedOn {
-            dependency = basedOn
-        }
-
+        
         // Inform the delegate.
         delegate.removing(repository: dependency.repository.url)
+        
+        // It refers to dependency which should be removed. 
+        // In the case of non editable packages
+        // it would reference to dependency, whereas in the case of editable package 
+        // it would reference to dependency.basedOn.
```

Comment:
I think we just say something short like: `// Compute the dependency which we need to remove.`